contracts/vault/src/execute.rs

```diff
@@ -61,224 +42,113 @@ pub fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg) -> S
-        }
-    }
-
-    // Remove specified operators
-    for addr in &remove {
-        let validated_addr = deps.api.addr_validate(addr)?;
-        operators.retain(|op| op != validated_addr);
-    }
-
-    // Update config
-    config.operators = operators;
-    state::CONFIG.save(deps.storage, &config)?;
-
-    // Return response
-    Ok(Response::new()
-        .add_attribute("action", "update_operators")
-        .add_attribute("added", add.len().to_string())
-        .add_attribute("removed", remove.len().to_string()))
-}
+fn execute_deposit(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Response> {
+    let config = state::CONFIG.load(deps.storage)?;
+    check_not_paused(&config)?;
+    let sender = info.sender.clone();
 
-/// Allows a user to deposit USDC and receive USDCLP tokens
-///
-/// # Parameters
-/// - `deps`: Mutable dependencies
-/// - `env`: Contract environment
-/// - `info`: Message information
-///
-/// # Returns
-/// - `StdResult<Response>`: Response with mint message and attributes
-fn execute_deposit(deps: DepsMut, env: Env, info: MessageInfo) -> StdResult<Response> {
-    check_not_paused(&deps.as_ref())?; // Ensure the contract is not paused
-    let config = state::CONFIG.load(deps.storage)?; // Load configuration
-    let sender = info.sender.clone(); // Address of the user depositing
-
-    // Get the amount of USDC sent
-    let usdc = info
-        .funds
-        .iter()
-        .find(|c| c.denom == config.usdc_denom)
-        .ok_or_else(|| StdError::generic_err("No USDC sent"))?;
-    let amount = usdc.amount; // Use the sent amount directly
-
-    if amount.is_zero() {
-        return Err(StdError::generic_err(
-            "Deposit amount must be greater than zero",
-        ));
+    if info.funds.len() != 1 || info.funds[0].denom != config.usdc_denom {
+        return Err(anyhow!("Exactly one coin (USDC) must be sent",));
     }
+    let amount = info.funds[0].amount;
 
     // Calculate the amount of USDCLP tokens to mint based on total assets and LP supply
     // Note: This assumes total_assets reflects the net value without impairment.
     // Future complexity (e.g., impairment, fees) may require adjusting total_assets or adding factors.
     let total_assets = get_total_assets(deps.as_ref(), &env)?;
     let total_lp = state::TOTAL_LP_SUPPLY.load(deps.storage)?;
     let lp_amount = if total_lp.is_zero() {
-        amount // Initial deposit: 1:1 ratio
+        amount
     } else {
-        amount.multiply_ratio(total_lp, total_assets.total_assets) // Proportional to existing LP
+        amount.multiply_ratio(total_lp, total_assets.total_assets)
     };
 
-    // Create a message to mint USDCLP tokens as CW20
-    let mint_msg = WasmMsg::Execute {
-        contract_addr: config.usdclp_address,
-        msg: to_json_binary(&Cw20ExecuteMsg::Mint {
-            recipient: sender.to_string(),
-            amount: lp_amount,
-        })?,
-        funds: vec![],
-    };
```

Comment:
You removed this and `Cw20ExecuteMsg::BurnFrom` below but did not replace them with any of the token factory messages that I proposed.

Reply:
Ok, so if I'm understanding correctly, minting/burning won't be necessary at all and these actions are adequately replaced by locking/unlocking according to the logic that the contract implements?